stdlib.sh

```diff
@@ -61,6 +61,113 @@ puts_verbose() {
   fi
 }
 
+# Key Value Store Utility
+# -------------------
+
+# This is a utility module that allows you to create a file that serves
+# as a key-value store that can be persisted to disk.
+#
+# It works around using lines formatted like:
+# 
+# key=value
+#
+# and using grep when searching for keys to find the last entry. This 
+# allows an existing key to be overwritten.
+#
+# This is inspired by the "world's simplest database" in Ch 3 of
+# Designing Data-Intensive Applications https://dataintensive.net/
+
+# Ensure that the file that we would like to persist this store to
+# exists. Also handles creating the containing directory if it doesn't
+# exist yet. 
+#
+# Will not overwrite the store if it already exists
+# Usage: kv_create file
+kv_create() {
+  local f=$1
+  mkdir -p "$(dirname "$f")"
+  touch "$f"
+}
+
+# Delete all keys and values from the store
+# Usage: kv_clear file
+kv_clear() {
+  local f=$1
+  echo "" > "$f"
+}
+
+# Set value for a key
+# Usage: kv_set file key value
+kv_set() {
+  if [[ $# -eq 3 ]]; then
+    local f=$1
+    if [[ -f $f ]]; then
+      echo "$2=$3" >> "$f"
+    fi
+  fi
+}
+
+# Retrieve value for a key. Will return an empty string if the key is
+# not present
+# Usage: kv_get file key
+kv_get() {
+  if [[ $# -eq 2 ]]; then
+    local f=$1
+    if [[ -f $f ]]; then
+      grep "^$2=" "$f" | sed -e "s/^$2=//" | tail -n 1
+    fi
+  fi
+}
+
+# Retrive the value for a key, but wrap it in quotes if the value contains spaces
+# This is expected to be mostly used internally
+# Usage: kv_get_escaped file key
+kv_get_escaped() {
```

Comment:
... and make this `kv_get`, which does `printf "%b\n" "$(kv_get_raw …)"` (the quoting might be off there)